src/vos/tests/vts_pm.c

```diff
@@ -1618,50 +1657,88 @@ minor_epoch_punch(void **state)
 	memset(buf, 'x', sizeof(buf));
-	d_iov_set(&sgl.sg_iovs[0], (void *)buf, sizeof(buf));
+	d_iov_set(&sgl.sg_iovs[0], (void *)buf, strlen(expected));
 	rc = vos_obj_fetch(arg->ctx.tc_co_hdl, oid, epoch++, 0, &dkey, 1,
 			   &iod, &sgl);
 	assert_int_equal(rc, 0);
 
-	assert_int_equal(iod.iod_size, 0);
-	assert_memory_equal(buf, sexpected, strlen(sexpected));
+	assert_memory_equal(buf, expected, strlen(expected));
 
-	/** Let's simulate replay scenario where rebuild replays a minor epoch
-	 * punch and an update with the same major epoch
-	 *
-	 * This can happen with something like the following
-	 * DTX does an update and commits at epoch 1.
-	 * DAOS takes a snapshot at epoch 2.
-	 * DTX does a distributed transaction that does punch and update at
-	 * 3.1 and 3.2, respectively and commits
-	 *
-	 * At some later time, rebuild runs.  While rebuilding the snapshot,
-	 * it will replay the future punch at 3.   Internally, on replay,
-	 * the minor epoch is set to MAX for updates and MAX - 1 for punch.
-	 */
+	daos_sgl_fini(&sgl, false);
+}
+
+static void
+minor_epoch_punch_rebuild(void **state)
+{
+	struct io_test_args	*arg = *state;
+	int			rc = 0;
+	daos_key_t		dkey;
+	daos_key_t		akey;
+	daos_recx_t		rex;
+	daos_iod_t		iod;
+	d_sg_list_t		sgl;
+	daos_epoch_t		epoch = 2000;
+	const char		*expected = "xxxxxlonelyworld";
+	const char		*first = "hello";
+	const char		*second = "lonelyworld";
+	char			buf[32] = {0};
+	char			dkey_buf[UPDATE_DKEY_SIZE];
+	char			akey_buf[UPDATE_DKEY_SIZE];
+	daos_unit_oid_t		oid;
+
+	test_args_reset(arg, VPOOL_SIZE);
+
+	memset(&rex, 0, sizeof(rex));
+	memset(&iod, 0, sizeof(iod));
+
+	/* set up dkey and akey */
+	oid = gen_oid(arg->ofeat);
+	vts_key_gen(&dkey_buf[0], arg->dkey_size, true, arg);
 	vts_key_gen(&akey_buf[0], arg->akey_size, false, arg);
+	set_iov(&dkey, &dkey_buf[0], arg->ofeat & DAOS_OF_DKEY_UINT64);
+	set_iov(&akey, &akey_buf[0], arg->ofeat & DAOS_OF_AKEY_UINT64);
+
+	rc = daos_sgl_init(&sgl, 1);
+	assert_int_equal(rc, 0);
+
+	rex.rx_idx = 0;
+	rex.rx_nr = strlen(first);
+
 	iod.iod_type = DAOS_IOD_ARRAY;
 	iod.iod_size = 1;
+	iod.iod_name = akey;
 	iod.iod_recxs = &rex;
 	iod.iod_nr = 1;
-	rex.rx_idx = 0;
-	rex.rx_nr = strlen(first);
 
+	/** let's simulate replay scenario where rebuild replays a minor epoch
+	 * punch and an update with the same major epoch
+	 *
+	 * this can happen with something like the following
+	 * dtx does an update and commits at epoch 1.
+	 * daos takes a snapshot at epoch 2.
+	 * dtx does a distributed transaction that does punch and update at
+	 * 3.1 and 3.2, respectively and commits
+	 *
+	 * at some later time, rebuild runs.  while rebuilding the snapshot,
+	 * it will replay the future punch at 3.   internally, on replay,
+	 * the minor epoch is set to max for updates and max - 1 for punch.
+	 */
 	d_iov_set(&sgl.sg_iovs[0], (void *)first, rex.rx_nr);
 	epoch++;
-	/** First write the punched extent */
+	/** first write the punched extent */
```

Comment:
Just curious what the convention is for upper vs lower case? There are several changes that change to lower case.

Reply:
I did something in vim that did this.   I reverted the ones needed to compile